### PR TITLE
Theia Generator

### PR DESCRIPTION
**Pull request by akosyakov**

### CODE REVIEW

File: examples/browser/package.json [hunk 1]
```diff
@@ -4,69 +4,57 @@
...
     "watch:localdeps": "ldm watch --sync --run=watch --original-sources",
     "test:ui": "wdio wdio.conf.js"
   },
   "devDependencies": {
-    "@types/chai": "^3.5.1",
-    "@types/webdriverio": "^4.7.0",
-    "@types/yargs": "^6.6.0",
-    "bunyan": "^1.8.10",
-    "chai": "^3.5.0",
+    "rimraf": "^2.6.1",
```
Comment (kittaakos) [inline examples/browser/package.json]: Please do not add the modules manually. It messes up the order.

Comment (akosyakov) [inline examples/browser/package.json]: This file is auto-generated, one should change `theia.package.json` and run `yo theia:browser --force`.

Comment (kittaakos) [inline examples/browser/package.json]: Would it be possible to generate a correct package.json? By correct, I meant the correct order of dependencies?

Comment (akosyakov) [inline examples/browser/package.json]: What is it? alphabetic?

Comment (kittaakos) [inline examples/browser/package.json]: Yes. Thanks for taking care of it!

File: config/generator-theia/src/browser/browser-frontend-generator.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    protected compileIndexHtml(frontendModules: Map<string, string>): string {
+        return `<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset="UTF-8">
+  <link href="http://maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css" rel="stylesheet">
+  <script type="text/javascript" src="https://www.promisejs.org/polyfills/promise-6.1.0.js" charset="utf-8"></script>
```
Comment (hexa00) [inline config/generator-theia/src/browser/browser-frontend-generator.ts]: Why do we need those external refs?

**Comment by hexa00** [comment]
@akosyakov  That's very nice, one thing however when you change code in the logger could you update Internals.md also ? It's out of date now...

File: config/generator-theia/src/browser/browser-frontend-generator.ts (hunk 2 above)
Comment (akosyakov) [inline config/generator-theia/src/browser/browser-frontend-generator.ts]: these is already an issue about it: https://github.com/theia-ide/theia/issues/251 someone should take care, should be pretty straightforward

**Comment by akosyakov** [comment]
I think i did not changed existing APIs of the logger, only added new methods. Could you point me to mentioned changes?

File: config/generator-theia/src/browser/browser-frontend-generator.ts (hunk 2 above)
Comment (hexa00) [inline config/generator-theia/src/browser/browser-frontend-generator.ts]: OK I'll try to add it to our next sprint, thx

**Comment by hexa00** [comment]
The logger-server-module.ts is now merged in logger-backend-module

**Comment by hexa00** [comment]
See : https://github.com/theia-ide/theia/blob/master/doc/Internals.md#registering-a-service
(Not a big change)

**Pull request merged**